contributing/chestertons-fence.md

```diff
@@ -0,0 +1,77 @@
+And remember, the more obvious your idea seems - the more likely its been discussed and debating many times over, so when you present it, present it with the potential caveats and pitfalls in place.
+If you have an instinct to say "why not **just** ...", then it almost certainly falls into this category.
+
+## Got an example?
+
+A common frustration is that students submit code to mentors that does not pass the tests.
+An equally common suggestion is "Why not just automatically run the tests in the CLI before someone submits their code?"
+It seems like a great idea - the CLI can just call a command to run the tests, and not let the student submit if they're broken.
+
+So firstly, what does it mean to "just call a command to run the tests".
+It means writing a script for each of the 52 languages on Exercism that can run the tests and check the outcome.
+That's a bit of effort, but doable.
+
+But it also means writing that script so that it runs on Windows, MacOSX and Linux, all in any possible version/configuration. That's a lot (if not an unbounded) amount of work.
+"Why does it have to be _every_ configuration?" you may ask?
+Well, because you're actively blocking someone from using Exercism unless they can run this script.
+That also means that it can never fail.
+If for some reason the script doesn't run, then the student is entirely blocked.
+A bug in one of those 52*3*n scripts means a student can't use the track any more on that OS.
+How do you test that?
+You can't.
+
+But, you say, you could just add a `--skip-tests` flag.
+Yes, you could - solid suggestion.
+But then, that means people can also choose to bypass the tests whenever they want anyway, which gets you back to the starting point, except now it's marginally less likely that broken tests are submitted, which means mentors have **less** expectation of that, meaning it's more jarring and confusing when the tests **are** broken.
+
+"But people wouldn't do that generally" you may argue. True, for some languages where it takes 0.5s to run the tests.
+But for languages that take 20s to run the tests, having to wait an extra 20s to submit is incredbily frustrating for students.
+
+So rather than posing the suggestion of "why not run the tests before submission", let's ask a question: "Why do people post solutions that don't pass the tests?"
+And then things get interesting.
+The general answer is because they're stuck or confused.
+And they need help.
+Which means that submitting broken tests is a really important heuristic for mentors that this student needs help.
+Yes, it's frustrating because mentors can't immediately know if the code is "correct" or not, but by then downloading the code and running it to check it (which most mentors to on non-trival solutions), they can know with 100% certainty whether the code is correct or not.
```

Comment:
```suggestion
Yes, it's frustrating because mentors can't immediately know if the code is "correct" or not unless they download the code and run it to verify it (which most mentors do on non-trivial solutions). Only then can they know with 100% certainty.
```

Reply:
```suggestion
Yes, it's frustrating because mentors don't immediately know if the code is "correct" or not. However, they can download the code and run it to verify it (which most mentors do on non-trivial solutions) and then know with 100% certainty whether it is correct or not.
```

@BethanyG I've reworded your solution slightly to try and keep my original intention of "this is a good thing" more clearly.

Reply:
@iHiD  I really like your edit.  I was having trouble with the wording, and that's much better!

Reply:
Great. Thanks! :)